torch_tvm/compiler.cpp

```diff
@@ -320,8 +342,22 @@ void TVMCompiler::run(Stack& stack) {
       value_to_ivalue[value] = optional_ivalue.value();
     }
     auto ivalue = value_to_ivalue.at(cache_[spec].input_values[i]);
-    auto tensor = ivalue.toTensor().to(at::kFloat);
-    auto dl_tensor = at::toDLPack(tensor);
+    //auto tensor = ivalue.toTensor().to(at::kFloat);
+    auto tensor = ivalue.toTensor();
+    DLManagedTensor* dl_tensor;
+    if (torch_tvm::utils::is_aligned(tensor.data_ptr(),
+          tvm::runtime::kAllocAlignment)) {
+      dl_tensor = at::toDLPack(tensor);
+    } else if (cache_[spec].tvm_param_tensors_.value_exists(value)) {
+      // Is the value parameter that is assumed to immutable?
+      dl_tensor = cache_[spec].tvm_param_tensors_.
+        get_tensor_like_for_value(value, tensor);
+    } else {
```

Comment:
isn't this final case unsound?

Reply:
We need this if the input data, not parameter, is unaligned.